Ускорить линтеры на CI (#22)

birthdaysgift opened the pull request:
Мы можем ускорить работу линтеров и пайтеста на CI. Для майпая, раффа и пайтеста можно добавить кэширование, а для пайлинта увеличить количество процессов.

В рамках этой задачи необходимо:
- добавить сохранение кэша майпая, раффа и пайтеста на GitHub
- добавить запуск на нескольких процессах для пайлинта

Rey-Lane commented on envs/ci/docker-compose.yml:
```diff
@@ -12,10 +12,16 @@ services:
     <<: *app-build
+    volumes:
+      # bind .mypy_cache to the host in order to store mypy cache in GitHub Cache
+      - ../../.mypy_cache:/forum123/.mypy_cache  # host path is relative to the current docker-compose file
```
Надо поменять путь, `forum123` явно отсылает на это)

birthdaysgift committed:
22: Ускорить линтеры на CI

Мы можем ускорить работу линтеров и пайтеста на CI. Для майпая, раффа и пайтеста можно добавить кэширование, а для пайлинта увеличить количество процессов.

В рамках этой задачи необходимо:

    - добавить сохранение кэша майпая, раффа и пайтеста на GitHub
    - добавить запуск на нескольких процессах для пайлинта

birthdaysgift commented:
Тип бенчмарки:
|  | Before | After |
|--------|--------|--------|
| Lint | ![image](https://user-images.githubusercontent.com/35929293/232316364-8cd058c1-9912-4bdf-a243-d9f3e8f212b2.png)| ![image](https://user-images.githubusercontent.com/35929293/232316462-68d06b5e-f04c-454e-a181-96f2cbcbe664.png) |
| Test | ![image](https://user-images.githubusercontent.com/35929293/232316404-0b24d43d-c422-4c75-a659-3f614d0c49ef.png) | ![image](https://user-images.githubusercontent.com/35929293/232316505-9553562e-302b-4783-96d5-5e9e845f6545.png) | 

Да, стало не намного быстрее, а где-то даже медленнее, но это связано с тем, что на данный момент у нас практически нет кода, который линтится и скорость работы линтеров/тестов на CI определяется только временем работы докера и временем загрузки кэша с гитхаба. В будущем, когда в репозитории будет больше кода, кэш будет играть более значительную роль и прирост в скорости будет более заметнее (я надеюсь).

birthdaysgift commented:
Коммит, на который ссылается пулл реквест не существует, так как мы меняли соглашение по именованию коммитов и ребейзили `develop` ветку. Подробнее об этом можно почитать в [обсуджении](https://github.com/week-password/wisher-backend/discussions/3#discussioncomment-5640598).

Хэш нового коммита, содержащего точную копию изменений внесённых этим пулл-реквестом: e4bd4ff
